### PR TITLE
fix detecting generated generic for dynamics

### PR DESCRIPTION
**Pull request by maks**
Should I also add a (integration?) test for this? as currently nothing picks up that this didn't actually change the generated code as expected.

Fixes: #89

### CODE REVIEW

**Comment by rrousselGit** [comment]
Yeah we'll need a test for this.

**Comment by maks** [comment]
@rrousselGit no worries, I'll add a test shortly

**Comment by Matsue** [comment]
@maks 
If it is not annoying you, please use those tests. Those tests are failed before this change and passed after this change. 

https://github.com/maks/freezed/pull/1

**Comment by maks** [comment]
Big thank you to @Matsue for adding tests for the fix.

File: packages/freezed/test/multiple_constructors_test.dart [hunk 1]
```diff
@@ -289,4 +279,57 @@ void main() {
...
+  group('NestedMap', () {
+    test('generates Map of correct type', () async {
+      final nestedMapClass = await _getClassElement('ShallowNestedMap');
+
+      expect(nestedMapClass.getField('children').type.getDisplayString(),
+          'Map<String, LeafNestedMapItem>');
+    });
+
+    test('generates Map of correct type for deeply nested case', () async {
+      final nestedMapClass = await _getClassElement('DeepNestedMap');
```
Comment (rrousselGit) [inline packages/freezed/test/multiple_constructors_test.dart:311]: _getClassElement is very expensive. It shouldn't be called more than once per test.

**Comment by rrousselGit** [comment]
Thanks!

**Pull request merged**